src/main/java/org/jenkinsci/plugins/workflow/support/actions/LogActionImpl.java

```diff
@@ -119,6 +142,20 @@ public FlowNode getParent() {
     }
 
     /**
+     * Unlike {@link FlowNode#isRunning}, handles {@link BlockStartNode}s.
+     * Note that {@code WorkflowRun.copyLogs} will still currently refuse to consider steps
+     * which fail to obtain a {@link TaskListener} prior to invoking their body
+     * (for example by calling {@link StepContext#get} on demand rather than by using {@link StepContextParameter}).
+     */
+    private static boolean isRunning(FlowNode node) {
+        if (node instanceof BlockStartNode) {
+            return new LinearBlockHoppingScanner().findFirstMatch(node.getExecution().getCurrentHeads(), Predicates.equalTo(node)) != null;
```

Comment:
> you are potentially scanning the flow back to the beginning of time

Right, because in

```groovy
node { // #1
  for (int i = 0; i < 1000; i++) {sh 'echo' /* #2 */}
}
```

the current head would include point 2 and we need to scan back to point 1 if this method is called. Now we could add caching etc. if there an API for it.